server/src/main/java/org/opensearch/monitor/fs/FsHealthService.java

```diff
@@ -126,14 +134,22 @@ public void setSlowPathLoggingThreshold(TimeValue slowPathLoggingThreshold) {
         this.slowPathLoggingThreshold = slowPathLoggingThreshold;
     }
 
+    public void setHealthyTimeoutThreshold(TimeValue healthyTimeoutThreshold) {
+        this.healthyTimeoutThreshold = healthyTimeoutThreshold;
+    }
+
     @Override
     public StatusInfo getHealth() {
         StatusInfo statusInfo;
         Set<Path> unhealthyPaths = this.unhealthyPaths;
-        if (enabled == false) {
+        if (enabled == false ) {
             statusInfo = new StatusInfo(HEALTHY, "health check disabled");
         } else if (brokenLock) {
             statusInfo = new StatusInfo(UNHEALTHY, "health check failed due to broken node lock");
+        } else if (lastRunTimeMillis.get() > Long.MIN_VALUE && currentTimeMillisSupplier.getAsLong() -
```

Comment:
This won't catch the if the first ever run for health check does not complete.

Reply:
Good point. Done!